In [12]:
import numpy as np
import pandas as pd
from sklearn import metrics

In [5]:
train_data = np.array(pd.read_csv('./data/train_data.csv'))

X = train_data[:, :-1]

y = train_data[:, -1]

In [6]:
X.shape

(5365, 80)

In [7]:
y.shape

(5365,)

In [39]:
#home team wins = 1
def predict_winner(X):
    '''Predicting the game winner only based on better team record. If both teams have the same record, predicts the home team as the winner.
    '''
    teamWin_index = 3
    teamLoss_index = 4
    opptWin_index = 43
    opptLoss_index = 44
    y_predicted = np.zeros(X.shape[0])
    for i, game in enumerate(X):
        awayWinPCT = game[teamWin_index] / (game[teamWin_index] + game[teamLoss_index])
        homeWinPCT = game[opptWin_index] / (game[opptWin_index] + game[opptLoss_index])
        if homeWinPCT >= awayWinPCT:
            y_predicted[i] = 1
    return y_predicted

Testing the *predict_winner()* method on train data.

In [40]:
y_predicted = predict_winner(X)

In [41]:
y_predicted.shape

(5365,)

In [42]:
metrics.confusion_matrix(y, y_predicted)

array([[1550,  664],
       [1171, 1980]])

In [43]:
metrics.accuracy_score(y, y_predicted)

0.6579683131407269

Testing the *predict_winner()* method on test data.

In [45]:
test_data = np.array(pd.read_csv('./data/test_data.csv'))

In [46]:
X_test = test_data[:, :-1]

In [47]:
y_test = test_data[:, -1]

In [48]:
y_test_predicted = predict_winner(X_test)

In [49]:
metrics.confusion_matrix(y_test, y_test_predicted)

array([[302, 147],
       [224, 402]])

In [50]:
metrics.accuracy_score(y_test, y_test_predicted)

0.6548837209302325

Choosing the team with the better record ahead of the NBA game gives us about 65.5% chance of being right. The goal will be to do better than 65.5%